src/GafferCyclesModule/GafferCyclesModule.cpp

```diff
@@ -453,6 +454,10 @@ BOOST_PYTHON_MODULE( _GafferCycles )
 
 	IECoreCycles::init();
 
+	// Must be initialized to ensure the statically linked Cycles module
+	// we are linked to can find the binaries for GPU rendering.
+	ccl::path_init( IECoreCycles::cyclesRoot() );
```

Comment:
In theory, _anything_ in `IECoreCycles::init()` could need this special duplicate treatment, right? And we're just lucky that at the moment there's just some additional logging bits and bobs in there?

This got me wondering - do we even need to link the GafferCycles module to Cycles at all? It already links to libGafferCycles, so can't it get the Cycles symbols from there? I gave that a go [here](https://github.com/johnhaddon/gaffer/commit/a849e911529e80d6f8090b183c9981f8ad6c7852) and it seemed to compile and run on Linux at least. Might be worth giving that a shot on Windows too? It'd certainly be less fragile if it works....

Reply:
We talked about this offline and tried a few things to remove the need for linking the library in both modules, but unfortunately none worked. Without the linking the Python module to the various Cycles libs, there are 11 unresolved symbols. We also tried linking GafferCycles with `/WHOLEARCHIVE` and variants on `/WHOLEARCHIVE:cycles_util` but none worked, instead generating different unresolved symbols within `GafferCycles`.

Leaving as-is for now, with the hope that some day there will be a dynamic Cycles library we can link to.